### PR TITLE
[custom_ops] make custom_ops.impl work on existing operators

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #106088
* __->__ #106076
* #106075

The design is that we construct a CustomOp object around the existing
operator and then use it to register things. It is totally OK if the
operator isn't functional (unlike torch._custom_ops.custom_op that can
only construct functional operators).

If the operator already has an implementation from a backend (either via
direct registration to e.g. DispatchKey::CPU, or an indirect
registration like CompositeImplicitAutograd/CompositeExplicitAutograd),
we raise an error.

Test Plan:
- new tests

### CODE REVIEW

**Comment by soulitzer** [comment]
> If the operator already has an implementation from a backend (either via
direct registration to e.g. DispatchKey::CPU, or an indirect
registration like CompositeImplicitAutograd/CompositeExplicitAutograd),
we raise an error.

This sounds good, but just curious - are there plans to also open it up to allow users to override existing implementations? Perhaps with an `override=True` flag? Or is that something where we tell users you should just use the lower-level torch library API instead.

File: torch/_custom_ops.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def impl(qualname, *, device_types=("cpu", "cuda"), func=None):
     """
 
     def inner(func):
-        custom_op = _find_custom_op(qualname)
+        custom_op = _find_custom_op(qualname, also_check_torch_library=True)
```
Comment (soulitzer) [inline torch/_custom_ops.py:172]: Maybe we want to advertise this new feature on the doc string?

**Review by soulitzer — APPROVED** [review]

**Comment by zou3519** [comment]
> This sounds good, but just curious - are there plans to also open it up to allow users to override existing implementations? Perhaps with an override=True flag? Or is that something where we tell users you should just use the lower-level torch library API instead.

override=True sounds reasonable to me

File: torch/_custom_ops.py (hunk 1 above)
Comment (zou3519) [reply torch/_custom_ops.py:172]: We could!

**Comment by zou3519** [comment]
@soulitzer I am going to merge this first without the override= option. The implementation warrants an additional PR, plus I want to see how people use this first without overide=.

**Pull request closed without merging**